### DISCUSSION TITLE
Asciidoctor-wasm

### DISCUSSION THREAD

**Discussion by tani** [ruby/ruby.wasm#470] [Show and tell]
Now I would like to announce the new implementation of Asciidoctor using ruby.wasm.

https://github.com/tani/asciidoctor-wasm . 

ruby.wasm is a pretty cool project. I would like to thank the developers!

**Comment by tani** [comment]
Through the release of this library, I have investigated an issue with `ruby.wasm`: its size. The `ruby.wasm` binary is currently quite large. To proceed with deploying to [jsr.io](https://jsr.io), we need to shrink the WASM binary to less than 20MB.

I believe we might be able to reduce the binary size by using the `-Os` compile flag when building `ruby.wasm`. If the developer (@kateinoigakukun?) can guide me on where to add the `-Os` flag in the build script, I will promptly create a pull request. Cheers!

**Comment by kateinoigakukun** [comment]
Thank you for trying ruby.wasm! 

You can try it by adding `crossruby.cflags = %w[-Os]` (or `-Oz` more aggressively?) here: https://github.com/ruby/ruby.wasm/blob/faa5102052ec265ed69fdbc91c812814487c97d2/lib/ruby_wasm/packager/core.rb#L310

(but as far as I remember, the size of the interpreter is not a big factor, but the size of .rb files of standard libraries is.)

**Reply by tani** [reply]
I see. I have the same result. Is there a way to compress the ruby module using zlib, zstd, etc?

**Reply by kateinoigakukun** [reply]
I expect such compression is already applied at http layer. To save storage space, you can manually apply compression by yourself before passing module to RubyVM JS API.
